Source/DataSource.swift

```diff
@@ -128,6 +128,34 @@ public struct DataSource<S: SectionInfoProtocol>: DataSourceProtocol {
     public func numberOfSections() -> Int {
         return sections.count
     }
+    
+    /**
+     Checks if a Section exists at a specific `IndexPath`.
+     
+     - parameter indexPath: The index path specifying the location of the cell.
+     - returns: `true` if the section exists at the specified `IndexPath`, otherwise it returns `false`.
+     */
+    public func sectionExists(for indexPath: IndexPath) -> Bool {
+        let section = indexPath.section
+        return section < numberOfSections()
+    }
+    
+    /**
+     Checks if an Item exists at a specific `IndexPath`.
+     
+     - parameter indexPath: The index path specifying the location of the cell.
+     - returns: `true` if the item exists at the specified `IndexPath`, otherwise it returns `false`.
+     */
+    public func itemExists(for indexPath: IndexPath) -> Bool {
+        if !sectionExists(for: indexPath) {
+            return false
+        }
+        
+        let section = indexPath.section
+        let row = indexPath.row
+        
+        return row < numberOfItems(inSection: section)
+    }
```

Comment:
I think we can get rid of both of these.

Instead, we can just check `if dataSource.item(atIndexPath: ip) == nil`